graphql/data/schema.js

```diff
@@ -1107,6 +1107,157 @@ const appType = new GraphQLObjectType({
 })
 
+//corresponds to UserMission table
+const UserMissionStats = new GraphQLObjectType({
+  name: 'UserMissionStats',
+  description: "an individual's stats for a mission",
+  fields: () => ({
+    missionId: {
+      type: GraphQLString,
+      description: 'Mission ID',
+    },
+    userId: {
+      type: GraphQLString,
+      description: "Users's userId",
+    },
+    username: {
+      type: GraphQLString,
+      description: "Users's username",
+    },
+    longestTabStreak: {
+      type: GraphQLInt,
+      description: 'the longest tab streak in days so far',
+    },
+    currentTabStreak: {
+      type: GraphQLInt,
+      description: 'the current tab streak in days so far',
+    },
+    tabCount: {
+      type: GraphQLInt,
+      description: "the user's' number of tabs towards mission",
+    },
+    missionMaxTabsDay: {
+      type: GraphQLInt,
+      description: 'the most tabs in a single day',
+    },
+    created: {
+      type: GraphQLString,
+      description: 'the timestamp from createSquadMutation',
+    },
+    acknowledgedMissionComplete: {
+      type: GraphQLBoolean,
+      description: 'if a user has acknowledged mission complete',
+    },
+    acknowledgedMissionStarted: {
+      type: GraphQLBoolean,
+      description: 'if a user has acknowledged mission started',
+    },
+  }),
+})
+
+const EndOfMissionAwards = new GraphQLObjectType({
+  name: 'EndOfMissionAwards',
+  description: 'persistant awards calculated at end of mission',
+  fields: () => ({
+    missionId: {
+      type: GraphQLString,
+      description: 'Mission ID',
+    },
+    mostConsistent: {
+      type: GraphQLString,
+      description: "Users's username who had the longest tab streak in days",
+    },
+    mostTabsInDay: {
+      type: GraphQLString,
+      description: "Users's username who had the most tabs in a day",
+    },
+    largestContributor: {
+      type: GraphQLInt,
+      description: "Users's username who had the most tabs overall",
+    },
+  }),
+})
+const getSquadMissionStats = () => {}
+const getCurrentUserMission = () => {}
+const getPastMissions = () => {}
+//mostly corresponds to Mission table, rolls up stats
+const MissionType = new GraphQLObjectType({
+  name: 'Mission',
+  description: 'the shape of a single mission',
+  fields: () => ({
+    missionId: {
+      type: GraphQLString,
+      description: 'Mission ID',
+    },
+    squadName: {
+      type: GraphQLString,
+      description: 'the name of the squad',
+    },
+    created: {
+      type: GraphQLString,
+      description: 'the timestamp from createSquadMutation',
+    },
+    started: {
+      type: GraphQLString,
+      description: 'the timestamp from first user accepting squad invite',
+    },
+    completed: {
+      type: GraphQLString,
+      description: 'the timestamp from when the mission completes',
+    },
+    tabGoal: {
+      type: GraphQLInt,
+      description: 'the number of tabs to complete mission',
+    },
+    missionType: {
```

Comment:
In addition, MissionType can and probably should include user-tailored individual mission data. E.g., the fields `acknowledgedMissionStarted` and `acknowledgedMissionComplete` can live here.